### ISSUE TITLE
Ignore usages that are part of feature test expressions

### ISSUE THREAD

**Issue by robatwilliams** [eslint-community/eslint-plugin-es-x#72]
If your code (or library code) is using a feature support test conditional statement before using a feature, then you still get an error reported by this plugin. It doesn't know your intent when doing such checks.

Could this plugin make guesses of reasonable confidence that statements like the following should not cause an error?

```javascript
if (String.prototype.replaceAll) {
   // use it
}
```

This issues arises from https://github.com/robatwilliams/es-compat/issues/72

**Comment by ota-meshi** [comment]
I think in order to reliably determine the method existence check, we need to analyze the code path and identify where the code is placed. I think it's very difficult to do, but if you have ideas please comment here or open a PR to add changes.

```js
if (String.prototype.replaceAll) {
   // use replaceAll
}
if (!String.prototype.replaceAll) {
   // use polyfill
} else {
   // use replaceAll
}
if (typeof String.prototype.replaceAll !== 'undefined') {
   // use replaceAll
}
if (typeof String.prototype.replaceAll === 'undefined') {
   // use polyfill
} else {
   // use replaceAll
}
a = String.prototype.replaceAll ? /* use replaceAll */ : /* use polyfill */;
```
